lessons/05_relax/03b_ConjugateGradient.ipynb

```diff
@@ -0,0 +1,541 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "##Example problem"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We thus consider again the Poisson equation\n",
+    "\n",
+    "\\begin{equation}\n",
+    "\\nabla^2 p = -2\\left(\\frac{\\pi}{2}\\right)^2\\sin\\left( \\frac{\\pi x}{L} \\right) \\cos\\left(\\frac{\\pi y}{L}\\right)\n",
+    "\\end{equation}\n",
+    "\n",
+    "in the domain \n",
+    "\n",
+    "$$\\left\\lbrace \\begin{align*}\n",
+    "0 &\\leq x\\leq 1  \\\\\n",
+    "-0.5 &\\leq y \\leq 0.5 \n",
+    "\\end{align*} \\right.$$\n",
+    "\n",
+    "with boundary conditions \n",
+    "\n",
+    "$$p=0 \\text{ at } \\left\\lbrace \n",
+    "\\begin{align*}\n",
+    "x&=0\\\\\n",
+    "y&=0\\\\\n",
+    "y&=-0.5\\\\\n",
+    "y&=0.5\n",
+    "\\end{align*} \\right.$$\n",
+    "\n",
+    "We will solve this equation by assuming an initial state of $p=0$ everywhere, and applying boundary conditions to relax in pseudo-time."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## A bit of theory"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The CG method aims at solving elliptic PDEs by iterating the unknown and have it converge, up to a given accuracy, to the solution dictated by the boundary conditions (it is a member of the so-called Krylov subspace methods, but this should not scare you...).\n",
+    "\n",
+    "Recall that in its discretized form, the Poisson equation reads,\n",
+    "\n",
+    "$$\\frac{p_{i+1,j}^{n}-2p_{i,j}^{n}+p_{i-1,j}^{n}}{\\Delta x^2}+\\frac{p_{i,j+1}^{n}-2 p_{i,j}^{n}+p_{i,j-1}^{n}}{\\Delta y^2}=b_{i,j}^{n}$$\n",
+    "\n",
+    "You may have not immediately noticed, but you certainly manipulated equations of this kind in the past. Indeed, the left hand side represents a linear combination of the values of $p$ at several grid points and this linear combination has to be equal to the value of the source term on the right hand side.\n",
+    "\n",
+    "Now imagine you gather the values $p_{i,j}$ of $p$ at all grid points into a big vector ${\\bf p}$ and you do the same for $b$ using the same ordering. Both vectors ${\\bf p}$ and ${\\bf b}$ contain $N=nx*ny$ values and thus belong to $\\mathbf{R}^N$. The linear relationship we just described means that the discretised Poisson equation can be symbolically written as,\n",
```

Comment:
Since we haven't mentioned Krylov subspace methods explicitly (except for the mention above) we should probably add an additional statement here about what R^N is.  Maybe by way of comparison to R^2, R^3, etc...